### PR TITLE
ref(remote-control): Use React/Redux.

### PR DESCRIPTION
**Pull request by hristoterezov**

### CODE REVIEW

File: modules/UI/UI.js [hunk 1]
```diff
@@ -305,45 +297,6 @@ UI.toggleFilmstrip = function() {
...
-        = `<input name="username" type="text"
-                placeholder="user@domain.net"
-                data-i18n="[placeholder]dialog.user"
-                class="input-control" autofocus>
-         <input name="password" type="password"
-                data-i18n="[placeholder]dialog.userPassword"
-                class="input-control"
-                placeholder="user password">`
-
-    ;
```
Comment (damencho) [inline modules/UI/UI.js:329]: Have you removed this? As people are using it and just submitted a PR fixing it https://github.com/jitsi/jitsi-meet/pull/8233

File: react/features/remote-control/middleware.js [hunk 2]
```diff
@@ -0,0 +1,92 @@
...
+        }
+
+        if (id === requestedParticipant) {
+            dispatch(clearRequest());
+            dispatch(setRemoteControlActive(false));
+        }
+
+        if (receiver?.controller === id) {
+            dispatch(stopReceiver(false, true));
+        }
```
Comment (damencho) [inline react/features/remote-control/middleware.js:85]: As I'm looking in some problems where we set some unnecessary states in the presence. Can you make sure that this does not do it either. For example you don't have it turned on initially and so there is no state in presence, but after some join/leave we push update in the presence with remote=off.

File: modules/UI/audio_levels/AudioLevels.js [hunk 3]
```diff
@@ -6,48 +6,6 @@ import UIUtil from '../util/UIUtil';
...
-     * indicated by "opacity".
-     *
-     * @param {string} elementID the parent audio indicator span element
-     * @param {number} index the index of the dots to fill, where 0 indicates
-     * the middle dot and the following increments point toward the
-     * corresponding pair of dots.
-     * @param {number} opacity the opacity to set for the specified dot.
-     */
-    _setDotLevel(elementID, index, opacity) {
-        let audioSpan
```
Comment (damencho) [inline modules/UI/audio_levels/AudioLevels.js:20]: This was not used?

File: modules/remotecontrol/Receiver.js [hunk 4]
```diff
@@ -1,331 +0,0 @@
...
-            logger.log('Remote control receiver disabled.');
-            this._stop(true);
-            APP.connection.removeFeature(DISCO_REMOTE_CONTROL_FEATURE);
-            APP.conference.removeConferenceListener(
-                JitsiConferenceEvents.ENDPOINT_MESSAGE_RECEIVED,
-                this._remoteControlEventsListener);
-            APP.conference.removeListener(
-                JitsiMeetConferenceEvents.BEFORE_HANGUP,
-                this._hangupListener);
-        }
```
Comment (damencho) [inline modules/remotecontrol/Receiver.js:110]: I don't see this addFeature and removeFeature in the PR, where are those?
Is there a case where the remote control can be disabled live, during a session?

Comment (hristoterezov) [reply modules/remotecontrol/Receiver.js:110]: This event was emitted by `hangup()` from conference.js. Now we directly dispatch `disableReceiver()` from there.

Comment (hristoterezov) [reply modules/remotecontrol/Receiver.js:110]: I can't think of a case when the the remote control will be disabled live. If such an use case appears we can add some logic to handle it. But currently the new code should be equivalent to the old one.

File: modules/UI/audio_levels/AudioLevels.js (hunk 3 above)
Comment (hristoterezov) [reply modules/UI/audio_levels/AudioLevels.js:20]: At least I wasn't able to find any usage of it...

File: modules/UI/UI.js (hunk 1 above)
Comment (hristoterezov) [reply modules/UI/UI.js:329]: We agreed offline that this is not used anymore.

File: react/features/remote-control/middleware.js (hunk 2 above)
Comment (hristoterezov) [reply react/features/remote-control/middleware.js:85]: I checked and the only place that would change the presence  is `setRemoteControlActive` action and it is called only at the beginning and the end of remote control session. It also checks if the value is changed before actually modifying the presence. We should be fine!

**Review by damencho — APPROVED** [review]

**Pull request merged**